crates/python/qcs_sdk/api.pyi

```diff
@@ -0,0 +1,326 @@
+"""
+The qcs_sdk module provides an interface to Rigetti Quantum Cloud Services. Allowing users to compile and run Quil programs on Rigetti quantum processors.
+"""
+from typing import Any, Dict, List, Optional, TypedDict
+from numbers import Number
+from enum import Enum
+
+from .qpu.client import QcsClient
+
+RecalculationTable = List[str]
+Memory = Dict[str, List[float]]
+PatchValues = Dict[str, List[float]]
+
+class ExecutionError(RuntimeError):
+    """Error encountered during program execution submission or when retrieving results."""
+    ...
+
+
+class TranslationError(RuntimeError):
+    """Error encountered during program translation."""
+    ...
+
+
+class CompilationError(RuntimeError):
+    """Error encountered during program compilation."""
+    ...
+
+
+class RewriteArithmeticError(RuntimeError):
+    """Error encountered rewriting arithmetic for program."""
+    ...
+
+
+class DeviceIsaError(ValueError):
+    """Error while building Instruction Set Architecture."""
+    ...
+
+
+class RewriteArithmeticResults(TypedDict):
```

Comment:
This class and below don't actually inherit from `TypeDict` (though we could update the macros to do that if we wanted to)